### DISCUSSION TITLE
Adding ICA mixing matrix to a different eeg_lst

### DISCUSSION THREAD

**Discussion by stonekate** [bnicenboim/eeguana#217] [Q&A] [answered]
Hey Bruno (or whoever's reading), I have some data where I need to run the ICA on data that's filtered differently to the data I want to do analysis on (long story but I can explain if it helps). It's apparently something that is normal to do - my colleague does it, although in ERPlab. In eeguana, I tried doing it something like:

```
dat1 <- eeg_filt_band_pass(dat, .freq = c(1, 30))

dat1_ica <-eeg_ica(dat1)

dat2 <- eeg_filt_band_pass(dat, .freq = c(0.01, 30))

dat2$.ica <- dat1$.ica

dat2 <- as_eeg_ica_lst(dat2) # error: could not find function "as_eeg_ica_lst"

dat2_icaed <- dat2 %>%  eeg_ica_keep(-13) %>%  as_eeg_lst() # error: no applicable method for 'eeg_ica_keep' applied to an object of class "eeg_lst"
```

I copied this out of your source code, but this last step doesn't work because `as_eeg_ica_lst()` doesn't seem to be a function. But then `eeg_ica_keep()` doesn't work if it's not an eeg_ica_lst. I can't find how `as_eeg_ica_lst()` is defined in your code. Anyway, any ideas on how I would do this if not this way? 

Thanks!

**Comment by bnicenboim** [comment, marked as answer]
yeah, the function exists but it's not exposed.
you can do:
```
eeguana:::as_eeg_ica_lst.eeg_lst(dat2)
```

I'll open an issue to make all this more straightforward with a `eeg_add_ica_mat()` function.

**Reply by stonekate** [reply]
It works! And it corrected the blinks nicely. Thanks!
